--- a/package/pkg-cargo.mk
+++ b/package/pkg-cargo.mk
@@ -153,15 +153,18 @@
 	CARGO_PROFILE_RELEASE_OVERFLOW_CHECKS="false"
 endif
 
-#
+# Isolate rustflags from host ~/.cargo/config.toml.
+PKG_CARGO_RUSTFLAGS = $(addprefix -C link-args=,$(TARGET_LDFLAGS))
+
 # This is a workaround for https://github.com/rust-lang/compiler-builtins/issues/420
 # and should be removed when fixed upstream
-#
 ifeq ($(NORMALIZED_ARCH),arm)
-	PKG_CARGO_ENV += \
-		CARGO_TARGET_$(call UPPERCASE,$(RUSTC_TARGET_NAME))_RUSTFLAGS="-Clink-arg=-Wl,--allow-multiple-definition"
+PKG_CARGO_RUSTFLAGS += -Clink-arg=-Wl,--allow-multiple-definition
 endif
 
+PKG_CARGO_ENV += \
+	RUSTFLAGS="$(PKG_CARGO_RUSTFLAGS)"
+
 HOST_PKG_CARGO_ENV = \
 	$(PKG_COMMON_CARGO_ENV) \
 	RUSTFLAGS="$(addprefix -C link-args=,$(HOST_LDFLAGS))"
