--- a/meson.build
+++ b/meson.build
@@ -255,7 +255,7 @@
 # Check if a minimal SpiderMonkey program compiles, links, and runs. If not,
 # it's most likely the case that SpiderMonkey was configured incorrectly, for
 # example by building mozglue as a shared library.
-minimal_program = cxx.run('''
+minimal_program_compiles = cxx.compiles('''
 #include <js/Initialization.h>
 int main() {
     if (!JS_Init()) return 1;
@@ -268,7 +268,7 @@

 recommended_configuration = ''' Check the recommended configuration:
 https://github.com/spidermonkey-embedders/spidermonkey-embedding-examples/blob/esr91/docs/Building%20SpiderMonkey.md'''
-if not minimal_program.compiled()
+if not minimal_program_compiles
     error('''A minimal SpiderMonkey program
 could not be compiled or linked. Most likely you should build it with a
 different configuration.''' + recommended_configuration)
@@ -277,10 +277,6 @@
 SpiderMonkey program executes will not be performed. Before shipping GJS, you
 should check that it does not crash on startup, since building SpiderMonkey with
 the wrong configuration may cause that.''' + recommended_configuration)
-elif minimal_program.returncode() != 0
-    error('''A minimal SpiderMonkey program
-failed to execute. Most likely you should build it with a different
-configuration.''' + recommended_configuration)
 endif

 ### Check for external programs ################################################
