--- a/src/libply-splash-core/ply-boot-splash.c
+++ b/src/libply-splash-core/ply-boot-splash.c
@@ -83,6 +83,7 @@
 
 static void ply_boot_splash_update_progress (ply_boot_splash_t *splash);
 static void ply_boot_splash_detach_from_event_loop (ply_boot_splash_t *splash);
+static void on_new_frame (ply_boot_splash_t *splash);
 
 ply_boot_splash_t *
 ply_boot_splash_new (const char   *theme_path,
@@ -401,6 +402,13 @@
                 return;
 
         if (splash->loop != NULL) {
+                if (splash->is_shown) {
+                        ply_event_loop_stop_watching_for_timeout (splash->loop,
+                                                                  (ply_event_loop_timeout_handler_t)
+                                                                  on_new_frame, splash);
+                        splash->is_shown = false;
+                }
+
                 if (splash->plugin_interface->on_boot_progress != NULL) {
                         ply_event_loop_stop_watching_for_timeout (splash->loop,
                                                                   (ply_event_loop_timeout_handler_t)
