--- a/meson_options.txt
+++ b/meson_options.txt
@@ -53,6 +53,11 @@
   value: true,
   description: 'Coordinate boot up with systemd',
 )
+option('systemd-ask-password-agent',
+  type: 'string',
+  value: '',
+  description: 'Target path to systemd-tty-ask-password-agent',
+)
 option('udev',
   type: 'feature',
   value: 'auto',
--- a/meson.build
+++ b/meson.build
@@ -44,7 +44,10 @@
   systemd_unit_dir = systemd_dep.get_variable('systemdsystemunitdir',
     pkgconfig_define: [ 'rootprefix', get_option('prefix') ],
   )
-  systemd_ask_password_agent = find_program('systemd-tty-ask-password-agent')
+  systemd_ask_password_agent = get_option('systemd-ask-password-agent')
+  if systemd_ask_password_agent == ''
+    systemd_ask_password_agent = find_program('systemd-tty-ask-password-agent').full_path()
+  endif
 endif
 
 if get_option('upstart-monitoring')
--- a/systemd-units/meson.build
+++ b/systemd-units/meson.build
@@ -21,7 +21,7 @@
       'PLYMOUTH_CLIENT_DIR': get_option('prefix') / get_option('bindir'),
       'PLYMOUTH_DAEMON_DIR': get_option('prefix') / get_option('sbindir'),
       'plymouthruntimedir': plymouth_runtime_dir,
-      'SYSTEMD_ASK_PASSWORD_AGENT': systemd_ask_password_agent.full_path(),
+      'SYSTEMD_ASK_PASSWORD_AGENT': systemd_ask_password_agent,
     },
     install: true,
     install_dir: systemd_unit_dir,
