--- a/src/camel/CMakeLists.txt	2025-01-03 20:00:44.000000000 +0200
+++ b/src/camel/CMakeLists.txt	2026-05-16 06:07:50.131982053 +0300
@@ -10,19 +10,27 @@
 	camel-url.h
 )
 
-add_executable(camel-gen-tables
-	camel-gen-tables.c
-)
+if(CMAKE_CROSSCOMPILING)
+	add_custom_command(
+		OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/camel-mime-tables.c
+		COMMAND ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/camel-mime-tables.c ${CMAKE_CURRENT_BINARY_DIR}/camel-mime-tables.c
+		DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/camel-mime-tables.c
+	)
+else()
+	add_executable(camel-gen-tables
+		camel-gen-tables.c
+	)
 
-target_include_directories(camel-gen-tables PUBLIC
-	${CMAKE_BINARY_DIR}
-)
+	target_include_directories(camel-gen-tables PUBLIC
+		${CMAKE_BINARY_DIR}
+	)
 
-add_custom_command(
-	OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/camel-mime-tables.c
-	COMMAND ${CMAKE_CURRENT_BINARY_DIR}/camel-gen-tables >${CMAKE_CURRENT_BINARY_DIR}/camel-mime-tables.c
-	DEPENDS camel-gen-tables
-)
+	add_custom_command(
+		OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/camel-mime-tables.c
+		COMMAND ${CMAKE_CURRENT_BINARY_DIR}/camel-gen-tables >${CMAKE_CURRENT_BINARY_DIR}/camel-mime-tables.c
+		DEPENDS camel-gen-tables
+	)
+endif()
 
 set(SOURCES
 	camel.c
--- a/src/addressbook/libebook-contacts/CMakeLists.txt	2025-01-03 20:00:44.000000000 +0200
+++ b/src/addressbook/libebook-contacts/CMakeLists.txt	2026-05-16 06:10:09.994954655 +0300
@@ -2,32 +2,40 @@
 
 add_pkgconfig_file(libebook-contacts.pc.in libebook-contacts-${API_VERSION}.pc)
 
-add_executable(gen-western-table
-	gen-western-table.c
-)
-
-target_compile_definitions(gen-western-table PRIVATE
-	-DG_LOG_DOMAIN=\"gen-western-table\"
-)
-
-target_compile_options(gen-western-table PUBLIC
-	${GNOME_PLATFORM_CFLAGS}
-)
-
-target_include_directories(gen-western-table PUBLIC
-	${CMAKE_BINARY_DIR}
-	${GNOME_PLATFORM_INCLUDE_DIRS}
-)
-
-target_link_libraries(gen-western-table
-	${GNOME_PLATFORM_LDFLAGS}
-)
-
-add_custom_command(
-	OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/e-name-western-tables.h
-	COMMAND ${CMAKE_CURRENT_BINARY_DIR}/gen-western-table "${CMAKE_CURRENT_SOURCE_DIR}/e-name-western-tables.h.in" >${CMAKE_CURRENT_BINARY_DIR}/e-name-western-tables.h
-	DEPENDS gen-western-table e-name-western-tables.h.in
-)
+if(CMAKE_CROSSCOMPILING)
+	add_custom_command(
+		OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/e-name-western-tables.h
+		COMMAND ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/e-name-western-tables.h ${CMAKE_CURRENT_BINARY_DIR}/e-name-western-tables.h
+		DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/e-name-western-tables.h
+	)
+else()
+	add_executable(gen-western-table
+		gen-western-table.c
+	)
+
+	target_compile_definitions(gen-western-table PRIVATE
+		-DG_LOG_DOMAIN="gen-western-table"
+	)
+
+	target_compile_options(gen-western-table PUBLIC
+		${GNOME_PLATFORM_CFLAGS}
+	)
+
+	target_include_directories(gen-western-table PUBLIC
+		${CMAKE_BINARY_DIR}
+		${GNOME_PLATFORM_INCLUDE_DIRS}
+	)
+
+	target_link_libraries(gen-western-table
+		${GNOME_PLATFORM_LDFLAGS}
+	)
+
+	add_custom_command(
+		OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/e-name-western-tables.h
+		COMMAND ${CMAKE_CURRENT_BINARY_DIR}/gen-western-table "${CMAKE_CURRENT_SOURCE_DIR}/e-name-western-tables.h.in" >${CMAKE_CURRENT_BINARY_DIR}/e-name-western-tables.h
+		DEPENDS gen-western-table e-name-western-tables.h.in
+	)
+endif()
 
 set(DEPENDENCIES
 	camel
