--- a/src/boot/dracut/module-setup.sh
+++ b/src/boot/dracut/module-setup.sh
@@ -24,7 +24,7 @@ installkernel() {
 }
 
 check() {
-    if [[ -x $systemdutildir/systemd ]] && [[ -x /usr/lib/ostree/ostree-prepare-root ]]; then
+    if [[ -x $dracutsysrootdir$systemdutildir/systemd ]] && [[ -x $dracutsysrootdir/usr/lib/ostree/ostree-prepare-root ]]; then
        return 255
     fi
 
@@ -38,11 +38,11 @@ depends() {
 install() {
     dracut_install /usr/lib/ostree/ostree-prepare-root
     for r in /usr/lib /etc; do
-        if test -f "$r/ostree/prepare-root.conf"; then
+        if test -f "$dracutsysrootdir$r/ostree/prepare-root.conf"; then
             inst_simple "$r/ostree/prepare-root.conf"
         fi
     done
-    if test -f "/etc/ostree/initramfs-root-binding.key"; then
+    if test -f "$dracutsysrootdir/etc/ostree/initramfs-root-binding.key"; then
         inst_simple "/etc/ostree/initramfs-root-binding.key"
     fi
     inst_simple "${systemdsystemunitdir}/ostree-prepare-root.service"
